### PR TITLE
build: make `cargo check` pass

### PR DESCRIPTION
**Pull request by piscisaureus**

### CODE REVIEW

File: msg_rs/build.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      src_root_dir.join("out").join(profile)
+    }
+  };
+
+  let msg_gen_rs = build_dir.join("gen/msg_generated.rs");
+  let msg_gen_rs = msg_gen_rs.to_string_lossy().replace("\\", "/");
+
+  let out_dir = env::var("OUT_DIR").unwrap();
+  let out_dir = Path::new(&out_dir);
+  let wrapper_rs = out_dir.join("find_msg_generated.inc.rs");
```
Comment (ry) [inline msg_rs/build.rs]: 1337

**Comment by ry** [comment]
I was able to fix this by doing this:
```
> DENO_BUILD_DIR=target/debug/ ./tools/build.py
> DENO_BUILD_DIR=target/debug/ cargo check
```

Please rename `msg_rs/cargo.toml` to `msg_rs/Cargo.toml`... 

Does msg_rs really need to be its own crate? This is something that can be dealt with in a follow up, but we should be able to just import it was as module...

**Comment by piscisaureus** [comment]
> I was able to fix this by doing this:
> 
> ```
> > DENO_BUILD_DIR=target/debug/ ./tools/build.py
> > DENO_BUILD_DIR=target/debug/ cargo check
> ```

That's expected -- build.rs makes no attempt to build or run flatc to generate `msg_generated.rs`. You have to do that manually before running `cargo check`.
This could be changed of course but that's for another day/PR.
PS: `DENO_BUILD_DIR` has no effect on anything.

> Please rename `msg_rs/cargo.toml` to `msg_rs/Cargo.toml`...

done

> Does msg_rs really need to be its own crate? This is something that can be dealt with in a follow up, but we should be able to just import it was as module...

I don't think so, necessarily. Although what's nice now is that this hackery is confined in it's own little box. Prefer to deal with this in a follow up PR indeed.

**Comment by piscisaureus** [comment]
@ry ptal

**Comment by ry** [comment]
@piscisaureus Please look at #1127 - upgrading flatbuffer coincidentally necessitated making msg_generated.rs a "mod" instead of a crate. I think we should land that first, since there's some complexity in your patch which will not exist #1127

File: .cargo/config [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+[build]
+# The location of the cargo build directory can be overridden, but the name of
+# the directory itself is best left at 'target'.
+# https://github.com/rust-lang/cargo/pull/1657#issue-36459101
+target-dir = "out/target"
```
Comment (ry) [inline .cargo/config]: If possible, let's leave this file out for now.

**Comment by piscisaureus** [comment]
Postponed to 2020.

**Comment by ry** [comment]
I'm resurrecting this in https://github.com/denoland/deno/pull/1128

**Pull request closed without merging**